### PR TITLE
docs: add description for contributing

### LINKED ISSUE #7: Documentation: add workflow for adding new profiler

### PR DESCRIPTION
**Pull request by Midnighter**
* [x] fix #7

I've added some contributing guidelines to the docs. We don't have rendering of the documentation set up yet so you will have to locally install the requirements and run `mkdocs serve` to see them.

Some of the links don't work yet but it's more important that the general description is ready, I think.

### CODE REVIEW

**Review by maxibor — APPROVED** [review]
Nice description, exactly what I was thinking of @Midnighter 💯💯

File: docs/contributing/index.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Contributing
+
+There are various ways for contributing to this project be that answering questions on the Slack channel or in GitHub issues, writing documentation and providing examples, testing the software in various settings, or submitting code through pull requests.
```
Comment (jfy133) [inline docs/contributing/index.md:3]:
```suggestion
There are various ways for contributing to this project be that answering questions on the nf-core [Slack channel](https://nfcore.slack.com/archives/C031QH57DSS) or in GitHub issues, writing documentation and providing examples, testing the software in various settings, or submitting code through pull requests.
```

File: docs/contributing/supporting_new_profiler.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Supporting New Taxonomic Profilers
+
+A good way to contribute to the taxpasta project, is to add support for a new taxonomic profiler. This mostly boils down to creating three new Python modules and filling them with life.
```
Comment (jfy133) [inline docs/contributing/supporting_new_profiler.md:3]:
```suggestion
A good way to contribute to the taxpasta project, is to add support for a new taxonomic profiler. This mostly boils down to creating three new Python files (modules) and filling them with life.
```

File: docs/contributing/supporting_new_profiler.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+```python
+from taxpasta.application import ProfileReader
+
+
+class Kraken2ProfileReader(ProfileReader):
+```
+
+## 2. Validate
+
+We are assuming that all profile data is tabular and therefore [pandas](https://pandas.pydata.org/) is our choice for handling this data. In order to perform validation of tabular data, we use [pandera schema models](https://pandera.readthedocs.io/en/stable/schema_models.html). The validation is automatically performed by decorating and annotating the `transform` classmethod of the standardisation service, for example,
```
Comment (jfy133) [inline docs/contributing/supporting_new_profiler.md:30]: Mention if not tabular, ask on the slack channel?

**Review by jfy133 — APPROVED** [review]
A couple of minor comments but I tihnk is relatively clear :)

**Review by maxibor — APPROVED** [review]

File: docs/contributing/supporting_new_profiler.md (hunk 3 above)
Comment (Midnighter) [reply docs/contributing/supporting_new_profiler.md:30]: Will add that 👍🏼

**Comment by Midnighter** [comment]
I made another few changes mainly describing testing but also changing the order. Please give it another look.

File: docs/contributing/supporting_new_profiler.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    def transform(
+        cls, profile: DataFrame[Kraken2Profile]
+    ) -> DataFrame[StandardProfile]:
+```
+
+The `pa.check_types` decorator validates the class method's input and output
+using the type annotations and the defined schema models.
+
+The `transform` class method itself needs to modify the given `pandas.DataFrame` such that
+the returned result looks like the [`StandardProfile`][].
```
Comment (jfy133) [inline docs/contributing/supporting_new_profiler.md:197]: Missing link? Expected for now?

**Review by jfy133 — APPROVED** [review]
Looks fine to me :+1:

File: docs/contributing/supporting_new_profiler.md (hunk 4 above)
Comment (Midnighter) [reply docs/contributing/supporting_new_profiler.md:197]: Ooops, missed that now. I'll fix it in a separate commit.

**Pull request merged**